### ISSUE TITLE
Error on standalone start

### ISSUE THREAD

**Issue by pontikos** [phenotips/phenotips#1951]
bash start.sh

Starting Jetty on port 8080, please wait...
2015-12-17 16:52:21,149 [main] WARN  o.a.s.c.SolrResourceLoader     - Can't find (or read) directory to add to classloader: lib (resolved as: /data/UCLpheno/distribution/standalone/target/phenotips-standalone-1.3-SNAPSHOT/data/solr/lib).
2015-12-17 16:52:22,452 [coreLoadExecutor-6-thread-1] ERROR o.a.s.c.CoreContainer          - Error creating core [omim]: Error opening new searcher
org.apache.solr.common.SolrException: Error opening new searcher
        at org.apache.solr.core.SolrCore.<init>(SolrCore.java:820) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.SolrCore.<init>(SolrCore.java:659) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.CoreContainer.create(CoreContainer.java:723) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.CoreContainer$1.call(CoreContainer.java:443) [solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.CoreContainer$1.call(CoreContainer.java:434) [solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at java.util.concurrent.FutureTask.run(FutureTask.java:262) [na:1.7.0_91]
        at org.apache.solr.common.util.ExecutorUtil$MDCAwareThreadPoolExecutor$1.run(ExecutorUtil.java:210) [solr-solrj-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:17]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_91]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_91]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_91]
Caused by: org.apache.solr.common.SolrException: Error opening new searcher
        at org.apache.solr.core.SolrCore.openNewSearcher(SolrCore.java:1662) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.SolrCore.getSearcher(SolrCore.java:1773) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.SolrCore.initSearcher(SolrCore.java:916) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.SolrCore.<init>(SolrCore.java:793) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        ... 9 common frames omitted
Caused by: org.apache.lucene.index.CorruptIndexException: codec footer mismatch (file truncated?): actual footer=609325604 vs expected footer=-1071082520 (resource=MMapIndexInput(path="/data/phenotips/data/solr/omim/data/index/_7.cfs"))
        at org.apache.lucene.codecs.CodecUtil.validateFooter(CodecUtil.java:416) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.codecs.CodecUtil.retrieveChecksum(CodecUtil.java:401) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.codecs.lucene50.Lucene50CompoundReader.<init>(Lucene50CompoundReader.java:79) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.codecs.lucene50.Lucene50CompoundFormat.getCompoundReader(Lucene50CompoundFormat.java:71) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.index.IndexWriter.readFieldInfos(IndexWriter.java:1015) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.index.IndexWriter.getFieldNumberMap(IndexWriter.java:1032) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:937) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.solr.update.SolrIndexWriter.<init>(SolrIndexWriter.java:78) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.update.SolrIndexWriter.create(SolrIndexWriter.java:65) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.update.DefaultSolrCoreState.createMainIndexWriter(DefaultSolrCoreState.java:274) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.update.DefaultSolrCoreState.getIndexWriter(DefaultSolrCoreState.java:117) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.core.SolrCore.openNewSearcher(SolrCore.java:1631) ~[solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        ... 12 common frames omitted
2015-12-17 16:52:23,442 [Finalizer] ERROR o.a.s.u.SolrIndexWriter        - SolrIndexWriter was not closed prior to finalize(), indicates a bug -- POSSIBLE RESOURCE LEAK!!!
2015-12-17 16:52:23,442 [Finalizer] ERROR o.a.s.u.SolrIndexWriter        - Error closing IndexWriter
java.lang.NullPointerException: null
        at org.apache.lucene.index.IndexWriter.doFlush(IndexWriter.java:3128) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.index.IndexWriter.flush(IndexWriter.java:3096) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.index.IndexWriter.shutdown(IndexWriter.java:1078) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.lucene.index.IndexWriter.close(IndexWriter.java:1123) ~[lucene-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:38:09]
        at org.apache.solr.update.SolrIndexWriter.close(SolrIndexWriter.java:130) [solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at org.apache.solr.update.SolrIndexWriter.finalize(SolrIndexWriter.java:183) [solr-core-5.3.1.jar:5.3.1 1703449 - noble - 2015-09-17 01:48:15]
        at java.lang.System$2.invokeFinalize(System.java:1213) [na:1.7.0_91]
        at java.lang.ref.Finalizer.runFinalizer(Finalizer.java:98) [na:1.7.0_91]
        at java.lang.ref.Finalizer.access$100(Finalizer.java:34) [na:1.7.0_91]
        at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:210) [na:1.7.0_91]

**Comment by Basi1i0** [comment]
Have same problem(

**Comment by pontikos** [comment]
I think my error was because I was out of space.  I'm running on a virtual machine and phenotips created large log files which filled up my disk space.  I think I still get these errors but the server runs now.

**Comment by Basi1i0** [comment]
Hmm. What is the approximate disk space size we are talking about? And thank you for the answer

**Comment by sdumitriu** [comment]
This error means that the solr resources aren't properly installed: the index files are corrupted, either truncated due to low disk space, as @pontikos is saying, or due to other file corruption reasons. Simply re-install the solr resources.

**Comment by sdumitriu** [comment]
@Basi1i0 Currently, those resources take about 350MB.

@pontikos Logs shouldn't be that big, unless you enabled debug logs or you keep getting the same stacktrace for every request, in which case you have a serious error that you should investigate.

**Comment by pontikos** [comment]
Sorry @sdumitriu how do fix the solr error?  Rebuild solr?  How exactly?
I've asked the same question on phenotips-users mailing list but I'm happy to post my question here instead.

**Comment by sdumitriu** [comment]
This only fixes the stacktrace with `Caused by: org.apache.lucene.index.CorruptIndexException: codec footer mismatch (file truncated?)`, not other warnings. To fix this:

- stop PhenoTips if running
- delete all the subdirectories in`/data/phenotips/data/solr/`
- get a clean copy of the solr configuration jar corresponding to your PhenoTips version from https://nexus.phenotips.org/nexus/content/repositories/releases/org/phenotips/solr-configuration/
- unzip it in a temporary place
- copy the result into `/data/phenotips/data/solr/`
- start PhenoTips

**Comment by pontikos** [comment]
Just to confirm that's the solr directory
in phenotips-standalone-1.3-SNAPSHOT/data ?

**Comment by sdumitriu** [comment]
Correct.

**Comment by pontikos** [comment]
Thank you that seems to have done it.
I'm now only getting this warning from solr:

```
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=192m; support was removed in 8.0
2016-02-05 16:48:09,096 [main] WARN  o.a.s.c.SolrResourceLoader     - Can't find (or read) directory to add to classloader: lib (resolved as: /home/rmhanpo/phenotips-standalone-1.3-SNAPSHOT/data/solr/lib).
2016-02-05 16:48:10,006 [coreLoadExecutor-6-thread-1] WARN  o.a.s.c.SolrCore               - [chebi] Solr index directory '/data/phenotipsdb/solr/chebi/data/index' doesn't exist. Creating new index...
2016-02-05 16:48:11,363 [coreLoadExecutor-6-thread-3] WARN  o.a.s.c.SolrCore               - [patients] Solr index directory '/data/phenotipsdb/solr/patients/data/index' doesn't exist. Creating new index...
```

But as you pointed out the ontologizer error that I raised on phenotips-users mailing list remains.
